js/modules/k6/experimental/fs/module.go

```diff
@@ -198,37 +198,32 @@ func (f *File) Read(into goja.Value) *goja.Promise {
 	promise, resolveFunc, rejectFunc := f.vu.Runtime().NewPromise()
 	callback := f.vu.RegisterCallback()
 
-	resolve := func(result any) {
-		callback(func() error {
-			resolveFunc(result)
-			return nil
-		})
-	}
-
-	reject := func(reason any) {
+	if common.IsNullish(into) {
 		callback(func() error {
-			rejectFunc(reason)
+			rejectFunc(newFsError(TypeError, "read() failed; reason: into cannot be null or undefined"))
 			return nil
 		})
-	}
-
-	if common.IsNullish(into) {
-		reject(newFsError(TypeError, "read() failed; reason: into cannot be null or undefined"))
 		return promise
 	}
 
 	// We expect the into argument to be a `Uint8Array` instance
 	intoObj := into.ToObject(f.vu.Runtime())
 	uint8ArrayConstructor := f.vu.Runtime().Get("Uint8Array")
 	if isUint8Array := intoObj.Get("constructor").SameAs(uint8ArrayConstructor); !isUint8Array {
-		reject(newFsError(TypeError, "read() failed; reason: into argument must be a Uint8Array"))
+		callback(func() error {
+			rejectFunc(newFsError(TypeError, "read() failed; reason: into argument must be a Uint8Array"))
+			return nil
+		})
```

Comment:
You don't need to use `callback` here you are already on the eventloop.

But you should move the `RegisterCallback` down to just before the `go func()` so it doesn't block

Reply:
nit: Also given the other changes I would rename `resolveFunc` to `resolve` and same for `rejectFunc`

Reply:
TIL! Wilco 👍🏻